packages/medusa/src/services/shipping-profile.ts

```diff
@@ -295,27 +302,22 @@ class ShippingProfileService extends TransactionBaseService {
 
       const { metadata, products, shipping_options, ...rest } = update
 
-      if (metadata) {
-        profile.metadata = setMetadata(profile, metadata)
-      }
-
       if (products) {
-        const productServiceTx = this.productService_.withTransaction(manager)
-        for (const pId of products) {
-          await productServiceTx.update(pId, {
-            profile_id: profile.id,
-          })
-        }
+        profile = await this.withTransaction(manager).addProduct(
+          profile.id,
+          products
+        )
       }
 
       if (shipping_options) {
-        const shippingOptionServiceTx =
-          this.shippingOptionService_.withTransaction(manager)
-        for (const oId of shipping_options) {
-          await shippingOptionServiceTx.update(oId, {
-            profile_id: profile.id,
-          })
-        }
+        profile = await this.withTransaction(manager).addShippingOption(
+          profile.id,
+          shipping_options
+        )
+      }
```

Comment:
todo: `withTransaction(manager)` is not required when using methods in the same class. At this point, the transaction manager will already be set and used in the class